query-service-impl/src/test/java/org/hypertrace/core/query/service/prometheus/QueryRequestToPromqlConverterTest.java

```diff
@@ -0,0 +1,217 @@
+    PromQLInstantQueries promqlQuery =
+        new QueryRequestToPromqlConverter(prometheusViewDefinition)
+            .convertToPromqlInstantQuery(
+                executionContext, builder.build(), createSelectionsFromQueryRequest(queryRequest));
+
+    // time filter is removed from the query
+    String query1 = "count by (service_name, api_name) (count_over_time(error_count{}[100ms]))";
+    String query2 = "avg by (service_name, api_name) (avg_over_time(num_calls{}[100ms]))";
+
+    Assertions.assertTrue(promqlQuery.getQueries().contains(query1));
+    Assertions.assertTrue(promqlQuery.getQueries().contains(query2));
+  }
+
+  @Test
+  void testInstantQueryWithGroupByWithMultipleAggregatesWithMultipleFilters() {
+    QueryRequest query = buildMultipleGroupByMultipleAggQueryWithMultipleFilters();
+    Builder builder = QueryRequest.newBuilder(query);
+    builder.setLimit(20);
+    PrometheusViewDefinition prometheusViewDefinition = getDefaultPrometheusViewDefinition();
+
+    QueryRequest queryRequest = builder.build();
+
+    ExecutionContext executionContext = new ExecutionContext("__default", queryRequest);
+    executionContext.setTimeFilterColumn("SERVICE.startTime");
+    PromQLInstantQueries promqlQuery =
+        new QueryRequestToPromqlConverter(prometheusViewDefinition)
+            .convertToPromqlInstantQuery(
+                executionContext, builder.build(), createSelectionsFromQueryRequest(queryRequest));
+
+    // time filter is removed from the query
+    String query1 =
+        "count by (service_name, api_name) (count_over_time(error_count{service_id=\"1|2|3\", service_name=~\"someregex\"}[100ms]))";
+    String query2 =
+        "avg by (service_name, api_name) (avg_over_time(num_calls{service_id=\"1|2|3\", service_name=~\"someregex\"}[100ms]))";
+
+    Assertions.assertTrue(promqlQuery.getQueries().contains(query1));
+    Assertions.assertTrue(promqlQuery.getQueries().contains(query2));
+  }
+
+  @Test
+  void testTimeSeriesQueryWithGroupByWithMultipleAggregatesWithMultipleFilters() {
+    QueryRequest query = buildMultipleGroupByMultipleAggQueryWithMultipleFiltersAndDateTime();
+    Builder builder = QueryRequest.newBuilder(query);
+    builder.setLimit(20);
+    PrometheusViewDefinition prometheusViewDefinition = getDefaultPrometheusViewDefinition();
+
+    QueryRequest queryRequest = builder.build();
+
+    ExecutionContext executionContext = new ExecutionContext("__default", queryRequest);
+    executionContext.setTimeFilterColumn("SERVICE.startTime");
+    PromQLRangeQueries promqlQuery =
+        new QueryRequestToPromqlConverter(prometheusViewDefinition)
+            .convertToPromqlRangeQuery(
+                executionContext, builder.build(), createSelectionsFromQueryRequest(queryRequest));
+
+    // time filter is removed from the query
+    String query1 =
+        "count by (service_name, api_name) (count_over_time(error_count{service_id=\"1|2|3\", service_name=~\"someregex\"}[100ms]))";
```

Comment:
I think, the value `100ms` of expression `[]`, we may have to change to value of time series period. we can talk on this.